packages/components/src/components/ui-shell/product-switcher.js

```diff
@@ -80,9 +82,7 @@ export default class ProductSwitcher extends NavigationMenuPanel {
     );
     const launcher = event.delegateTarget;
     if (!launcher.id) {
-      launcher.id = `__carbon-product-switcher-launcher-${Math.random()
-        .toString(36)
-        .substr(2)}`;
+      launcher.id = `__carbon-product-switcher-launcher-${seq++}`;
```

Comment:
FYI, id's must start with a letter in order to be valid HTML.  If this is an id that get's added to the HTML I would suggest removing the underscores.

Reply:
Given the point seems for older browsers probably we'd go for it as-is for now.